tools/entrypoint.sh

```diff
@@ -39,7 +39,8 @@ wdir="$(pwd)"
 if ! su-exec "$USERID" "$BASHPATH" -c "test -w $wdir && test -r $wdir"; then
   echo_error_and_exit "uid:gid $USERID lacks permissions to $wdir/"
 fi
-wdirgitindex="$wdir/.git/index"
+wdirgitdir="$(git rev-parse --git-dir)" || echo_error_and_exit "Cannot determine Git directory"
+wdirgitindex="${wdirgitdir}/index"
```

Comment:
Maybe re-use error from GIT CLI? 🤔 
```suggestion
wdirgitindex="$(git rev-parse --git-dir 2>&1)/index" || echo_error_and_exit "${wdirgitindex%/index}"
```